### ISSUE TITLE
research: distill a telemetry supervisor teacher into a small Spikenaut student

### ISSUE THREAD

**Issue by rmems** [rmems/SynapticDistill.jl#28]
## Planning attribution

**GPT-5.6 (high)**

## Context

`SynapticDistill.jl` is already designed for hybrid teacher→student SNN training:

- caller supplies the model step
- caller supplies the loss
- teacher-specific execution stays outside the package
- only the SNN student is updated
- e-prop / OTTT remain available as online/event-based learning rules

That architecture is a good fit for a new experiment: distill a larger telemetry supervisor (for example a `LiquidCortex.jl` reservoir/readout, a small ANN, or a frozen language/model-derived policy) into a compact `Spikenaut-SNN` student that can later be evaluated in software and on FPGA.

This issue should preserve package generality. The package should gain a reusable **supervisor-distillation example/contract**, not a hard dependency on any one teacher implementation.

## Research hypothesis

A compact SNN may preserve enough of a larger temporal supervisor's decision behavior to act as a low-cost always-on controller, while the larger teacher is needed only for offline labeling/distillation or occasional slow-loop guidance.

## Goal

Add a reproducible teacher-target → SNN-student distillation workflow for temporal supervisor tasks.

## Required input contract

Define a small application-neutral record type or documented schema for a sequence containing:

- spike/input tensor or feature sequence
- teacher target(s)
- optional teacher confidence/logits
- optional sample/session id
- optional reward / outcome metadata

Possible conceptual shape:

```julia
SupervisorTarget(
    sample_id,
    teacher_output,
    teacher_confidence,
    metadata,
)
```

Do not require a live teacher process during training. Precomputed teacher targets should be a first-class path so experiments are reproducible.

## Target modes

Support at least two target styles in examples/tests:

### 1. Continuous state distillation

Student predicts continuous teacher state such as:

```text
thermal_pressure
resource_stress
available_capacity
future_temperature
future_vram_pressure
```

Loss: MSE/Huber or caller-provided equivalent.

### 2. Discrete supervisor action distillation

Student predicts action/logit targets such as:

```text
ALLOW
WARN
THROTTLE
PAUSE
```

Loss: caller-provided cross-entropy / KL / margin loss.

The core package must not hard-code those labels; they belong in the example layer.

### Phase A — offline distillation

- teacher targets are frozen and precomputed
- student trains with deterministic replay
- no online weight update during evaluation

This is the required first result.

### Phase B — online student adaptation

After offline behavior is reproducible, allow the SNN student to continue adapting via e-prop or OTTT on streaming samples.

Requirements:

- explicit learning cadence
- explicit max update magnitude / clipping if applicable
- weights checkpointed before and after session
- online adaptation can be disabled for evaluation

### Phase C — teacher refresh / slow-loop guidance

Optional later experiment where a large teacher periodically emits new targets while the SNN handles the fast loop.

Do not implement this until A/B are stable.

## Evaluation

For held-out sessions report:

- teacher/student action agreement
- macro F1 for discrete actions
- false-safe rate (`student=ALLOW` when teacher/deterministic safety target=`PAUSE`)
- continuous target MAE/RMSE where applicable
- early-warning lead-time difference
- student spike rate
- student parameter count / memory footprint
- training time
- inference latency

When comparing to FPGA later, preserve the exact same evaluation records.

## Dataset discipline

- [ ] split by source session/run, not random adjacent rows
- [ ] persist dataset/target manifest and hash
- [ ] persist teacher identity/version/config
- [ ] persist teacher output normalization / temperature if used
- [ ] do not leak held-out outcomes into teacher targets for current-state decisions

## Suggested example

Add something like:

```text
examples/supervisor_distillation.jl
examples/data/supervisor_tiny_fixture.jsonl
```

The fixture should be synthetic and tiny.

The example should demonstrate:

```text
precomputed teacher targets
        ↓
spike/input sequence
        ↓
student model_step
        ↓
caller-defined distillation loss
        ↓
train_step!(...; rule=:eprop or :ottt)
        ↓
metric summary
```

## Package/API constraints

- [ ] no direct dependency on `LiquidCortex.jl`
- [ ] no direct dependency on `Spikenaut-SNN` repository layout
- [ ] no process/IPC requirement in the core package
- [ ] teacher remains frozen by construction in the example
- [ ] existing pure-SNN API remains unchanged unless a small generic addition is justified

## Tests

- [ ] precomputed target fixture trains deterministically with fixed seed
- [ ] injected model step is actually used
- [ ] only student parameters change
- [ ] missing/malformed teacher target fails clearly
- [ ] continuous-target example loss decreases on a trivial learnable fixture
- [ ] discrete-target example improves agreement above random baseline on a trivial fixture
- [ ] online adaptation can be fully disabled for held-out evaluation

## Acceptance criteria

A cold agent can run one documented Julia command and obtain:

1. a loaded synthetic teacher-target dataset,
2. an SNN student training run,
3. before/after loss,
4. teacher/student agreement metrics,
5. a saved student artifact or clearly documented parameter export hook.

## Non-goals

- Running Grok/Claude/etc. directly inside this package
- Host process control
- FPGA programming
- Claiming a distilled student is equivalent to the teacher
- Live financial trading

## Cross-repo research path

Expected research flow:

```text
hardware/market telemetry
        ↓
large teacher (e.g. LiquidCortex / ANN / frozen model policy)
        ↓
precomputed teacher targets
        ↓
SynapticDistill.jl
        ↓
Spikenaut-SNN student
        ↓
silicon-bridge / silicon-hdl parity experiments
```

Suggested labels: `research`, `distillation`, `neuromorphic`, `online-learning`.

**Comment by rmems** [comment]
Not this ticket. Script-level `spikenaut_train.jl` weight-quality fix is [#29](https://github.com/rmems/SynapticDistill.jl/issues/29) (related consumer [Spikenaut-SNN#13](https://github.com/rmems/Spikenaut-SNN/issues/13)). #28 stays teacher→student distillation API.
